types/jest/index.d.ts

```diff
@@ -466,3 +702,369 @@ declare namespace jasmine {
+
+    interface GlobalConfig {
+        bail: boolean;
+        collectCoverage: boolean;
+        collectCoverageFrom: Glob[];
+        collectCoverageOnlyFrom: void | {[key: string]: boolean};
+        coverageDirectory: string;
+        coverageReporters: string[];
+        coverageThreshold: {global: {[key: string]: number}};
+        expand: boolean;
+        forceExit: boolean;
+        logHeapUsage: boolean;
+        mapCoverage: boolean;
+        noStackTrace: boolean;
+        notify: boolean;
+        projects: Glob[];
+        replname: void | string;
+        reporters: ReporterConfig[];
+        rootDir: Path;
+        silent: boolean;
+        testNamePattern: string;
+        testPathPattern: string;
+        testResultsProcessor: void | string;
+        updateSnapshot: SnapshotUpdateState;
+        useStderr: boolean;
+        verbose: void | boolean;
+        watch: boolean;
+        watchman: boolean;
+    }
+
+    interface ProjectConfig {
+        automock: boolean;
+        browser: boolean;
+        cache: boolean;
+        cacheDirectory: Path;
+        clearMocks: boolean;
+        coveragePathIgnorePatterns: string[];
+        globals: ConfigGlobals;
+        haste: HasteConfig;
+        moduleDirectories: string[];
+        moduleFileExtensions: string[];
+        moduleLoader: Path;
+        moduleNameMapper: Array<[string, string]>;
+        modulePathIgnorePatterns: string[];
+        modulePaths: string[];
+        name: string;
+        resetMocks: boolean;
+        resetModules: boolean;
+        resolver: void | Path;
+        rootDir: Path;
+        roots: Path[];
+        setupFiles: Path[];
+        setupTestFrameworkScriptFile: Path;
+        skipNodeResolution: boolean;
+        snapshotSerializers: Path[];
+        testEnvironment: string;
+        testMatch: Glob[];
+        testPathIgnorePatterns: string[];
+        testRegex: string;
+        testRunner: string;
+        testURL: string;
+        timers: 'real' | 'fake';
+        transform: Array<[string, Path]>;
+        transformIgnorePatterns: Glob[];
+        unmockedModulePathPatterns: void | string[];
+    }
+
+    // Console
+
+    type LogMessage = string;
+    interface LogEntry {
+        message: LogMessage;
+        origin: string;
+        type: LogType;
+    }
+    type LogType = 'log' | 'info' | 'warn' | 'error';
+    type ConsoleBuffer = LogEntry[];
+
+    // Context
+
+    interface Context {
+        config: ProjectConfig;
+        hasteFS: HasteFS;
+        moduleMap: ModuleMap;
+        resolver: HasteResolver;
+    }
+
+    // Environment
+
+    interface FakeTimers {
+        clearAllTimers(): void;
+        runAllImmediates(): void;
+        runAllTicks(): void;
+        runAllTimers(): void;
+        runTimersToTime(msToRun: number): void;
+        runOnlyPendingTimers(): void;
+        runWithRealTimers(callback: any): void;
+        useFakeTimers(): void;
+        useRealTimers(): void;
+    }
+
+    interface $JestEnvironment {
```

Comment:
It's the name from jest repo, I think we should not change it for better comparing in the future.